src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -734,130 +740,714 @@ public object InvokeFunction(string methodName, Query query, string documentId,
+                                if (src == val)
+                                {
+                                    result = true;
+                                    break;
+                                }
+                            }
                         }
-                        count += span[0].Count;
+
+                        return result;
+                    }
+                    catch (Exception e)
+                    {
+                        throw new InvalidQueryException($"Time series function '{declaredFunction.Name}' failed to execute InExpression '{inExpression}' on : source '{src}', value '{val}'", e);
                     }
                 }
-            }
 
-            if (aggStates[0].Count > 0)
-            {
-                array.Add(AddTimeSeriesResult(func, aggStates, start, next));
-            }
+                bool EvaluateBetweenExpression()
+                {
+                    var result = false;
 
-            return _context.ReadObject(new DynamicJsonValue
-            {
-                ["Count"] = count,
-                ["Results"] = array
-            }, "timeseries/value", BlittableJsonDocumentBuilder.UsageMode.None);
+                    dynamic src = GetValue(betweenExpression.Source, singleResult);
+                    dynamic value = GetValue(betweenExpression.MinExpression, singleResult);
 
+                    try
+                    {
+                        if (src is LazyNumberValue lnv)
+                        {
+                            if (lnv.CompareTo(value) >= 0)
+                            {
+                                value = GetValue(betweenExpression.MaxExpression, singleResult);
+                                result = lnv.CompareTo(value) <= 0;
+                            }
+                        }
 
-            void AggregateIndividualItems(IEnumerable<SingleResult> items)
+                        else if (src >= value)
+                        {
+                            value = GetValue(betweenExpression.MaxExpression, singleResult);
+                            result = src <= value;
+                        }
+
+                        return result;
+                    }
+                    catch (Exception e)
+                    {
+                        throw new InvalidQueryException($"Time series function '{declaredFunction.Name}' failed to execute BetweenExpression '{betweenExpression}'", e);
+                    }
+
+                }
+            }
+
+
+            object GetValue(QueryExpression expression, SingleResult singleResult)
             {
-                foreach (var cur in items)
+                if (expression is FieldExpression fe)
                 {
-                    MaybeMoveToNextRange(cur.TimeStamp);
-                    count++;
-                    for (int i = 0; i < aggStates.Length; i++)
+                    switch (fe.Compound[0].Value.ToLower())
```

Comment:
Note, this does allocation per call. Can we avoid this?

Reply:
done